### PR TITLE
#11323: Enable all UNet Shallow tests in CI

### PR DESCRIPTION
**Pull request by esmalTT**
### Ticket
#11323

### What's changed
- Enables a functional test in the nightly pipeline, as well as end-to-end and device performance tests in their corresponding pipelines.
- Removed an old performance benchmarking script that is disabled, not mentioned in any CI pipelines, and no longer works, in favour of the one in `models/experimental`.
- Added some additional asserts to ensure users are running the correct configuration before using the model.
- Removed a bunch of unused imports.

### Checklist
- [x] Post commit CI passes
- [x] Model regression CI testing passes (if applicable)
  - [x] e2e perf -> https://github.com/tenstorrent/tt-metal/actions/runs/10392607624/job/28778794155
  - [x] device perf -> https://github.com/tenstorrent/tt-metal/actions/runs/10392606514/job/28778899256
  - [x] nightly -> https://github.com/tenstorrent/tt-metal/actions/runs/10388231257/job/28768106530, https://github.com/tenstorrent/tt-metal/actions/runs/10394106478

### CODE REVIEW

**Review by tt-rkim — APPROVED** [review]

**Comment by tt-rkim** [comment]
Merge once all pass! Should make sure the single card device and model perf pipelines actually run the test as well

**Review by yan-zaretskiy — APPROVED** [review]

File: models/experimental/functional_unet/unet_utils.py [hunk 1]
```diff
@@ -6,17 +6,18 @@
...
 from ttnn.model_preprocessing import preprocess_model, preprocess_conv2d, fold_batch_norm2d_into_conv2d
 
 from models.experimental.functional_unet.tt import unet_shallow_torch
 from models.experimental.functional_unet.tt import unet_shallow_ttnn
 
 
 def create_unet_models(device, groups, torch_input_tensor):
+    assert (device.core_grid.x, device.core_grid.y) == (
+        8,
+        8,
```
Comment (mywoodstock) [inline models/experimental/functional_unet/unet_utils.py:18]: should it be checking if the arch is WHB0?

Comment (mywoodstock) [reply models/experimental/functional_unet/unet_utils.py:18]: And may be we can extract the grid from device and then check if they are 8x8.

**Review by mywoodstock — APPROVED** [review]
minor comment

**Pull request merged**